File: backend/app/schemas/api_key.py
# Copyright (c) 2026 Valaris Studio
# SPDX-License-Identifier: AGPL-3.0-or-later

import uuid
from datetime import datetime

from app.core.json_response import UTCModel


class ApiKeyCreate(UTCModel):
    name: str


class ApiKeyRead(UTCModel):
    id: uuid.UUID
    name: str
    key_prefix: str
    created_at: datetime
    last_used_at: datetime | None = None

    model_config = {"from_attributes": True}


class ApiKeyCreated(ApiKeyRead):
    raw_key: str

File: frontend/src/types/api-key.ts
// Copyright (c) 2026 Valaris Studio
// SPDX-License-Identifier: AGPL-3.0-or-later

export interface ApiKey {
  id: string;
  name: string;
  key_prefix: string;
  created_at: string;
  // Null until the key's first authenticated call. The MCP connection wizard
  // reads this to decide whether a user still needs to be walked through the
  // connection or has already made it.
  last_used_at: string | null;
}

export interface ApiKeyCreated extends ApiKey {
  raw_key: string;
}
